emulate/emulate.js

```diff
@@ -0,0 +1,103 @@
+const redis = require("redis");
+const redisClient = redis.createClient("redis://redis:6379");
+const { NodeVM } = require("vm2");
+
+// https://www.npmjs.com/package/vm2
+
+redisClient.on("error", function (error) {
+  console.error(error);
+});
+
+/// Handles switching DMX values for the next frame
+function _NextFrame(dmxValuesIn) {
+  // TODO - Check if dmxValuesIn is correct
+  redisClient.set("DMXvalues", dmxValuesIn.toString());
+  redisClient.set("DMXvalues_update_timestamp", Date.now().toString());
+}
+
+/// Handles initializing the array with DMX values
+function _InitValues() {
+  let values = [];
+  for (let i = 0; i < 5; i++) {
+    let row = [];
+    for (let j = 0; j < 28; j++) {
+      row.push([0, 0, 0]);
+    }
+    values.push(row);
+  }
+  return values;
+}
+
+/// Handles errors from inside code emulation
+function _GetError(message) {
+  console.error(message);
+}
+
+/// Enables us to stop code emulation from the outside
+let working = true;
+function _IsWorking() {
+  return working;
+}
+
+let vm = new NodeVM({
+  timeout: 5,
+  sandbox: {
+    _IsWorking,
+    _NextFrame,
+    _InitValues,
+    _GetError,
+  },
+  console: "redirect",
+});
+
+vm.on("console.log", (data) => {
+  console.log(data);
+});
+
+// TODO - User code should be fetched from queue instead
+const UserCode = `
```

Comment:
zmienna nigdy z dużej.

tutaj i tak nie powinno być przykladu kodu, tylko brac z parametrów processu (stdin albo coś)

Reply:
To jest const, to nawet powinno być USERCODE najlepiej jak dla mnie, nie?

Reply:
Znaczy docelowo do `const` nie będzie tylko jak mówisz, brane z zewnątrz (stdin, pipe, plik, cokolwiek), a ta zmienna jest przejściowa, jak w TODO nad nią

Reply:
https://stackoverflow.com/questions/40291766/naming-convention-for-const-object-keys-in-es6/40291962

Bardziej chodziło mi, że jak robić CamelCase to nigdy z dużej na początku. Taka jest konwecja dla klas